services/kms/pkg/resource/key/sdk.go

```diff
@@ -0,0 +1,373 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package key
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"strings"
+
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	ackcompare "github.com/aws/aws-controllers-k8s/pkg/compare"
+	ackerr "github.com/aws/aws-controllers-k8s/pkg/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/kms"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws/aws-controllers-k8s/services/kms/apis/v1alpha1"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &metav1.Time{}
+	_ = strings.ToLower("")
+	_ = &aws.JSONValue{}
+	_ = &svcsdk.KMS{}
+	_ = &svcapitypes.Key{}
+	_ = ackv1alpha1.AWSAccountID("")
+	_ = &ackerr.NotFound
+)
+
+// sdkFind returns SDK-specific information about a supplied resource
+func (rm *resourceManager) sdkFind(
+	ctx context.Context,
+	r *resource,
+) (*resource, error) {
+	// If any required fields in the input shape are missing, AWS resource is
+	// not created yet. Return NotFound here to indicate to callers that the
+	// resource isn't yet created.
+	if rm.requiredFieldsMissingFromReadOneInput(r) {
+		return nil, ackerr.NotFound
+	}
+
+	input, err := rm.newDescribeRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, respErr := rm.sdkapi.DescribeKeyWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_ONE", "DescribeKey", respErr)
+	if respErr != nil {
+		if awsErr, ok := ackerr.AWSError(respErr); ok && awsErr.Code() == "UNKNOWN" {
+			return nil, ackerr.NotFound
+		}
+		return nil, respErr
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := r.ko.DeepCopy()
+
```

Comment:
I had to remove the following lines
```
-       if resp.KeyMetadata.CreationDate != nil {
-               ko.Status.CreationDate = &metav1.Time{*resp.KeyMetadata.CreationDate}
-       }
-       if resp.KeyMetadata.CustomerMasterKeySpec != nil {
-               ko.Spec.CustomerMasterKeySpec = resp.KeyMetadata.CustomerMasterKeySpec
-       }
-       if resp.KeyMetadata.Description != nil {
-               ko.Spec.Description = resp.KeyMetadata.Description
-       }
```

Reconcile was complaining about the diffs produced between desired state.
```
"diff": "\"Spec.CustomerMasterKeySpec\"\n\n\"Spec.KeyUsage\"\n\n\"Spec.Origin\"\n\n\"Status.CreationDate\"\n"
```

The heading of this file mentions that this file should not be edited, in which case where can I make some of these changes?

Reply:
@arajakul yeah, we're working on some changes to the code generator to handle certain field comparisons better. You've run into a great example of where the generated code falls down. You are correct that if you remove these lines of code, they will be re-inserted into `sdk.go` the next time you regenerate the controller. That said, there's nothing stopping you from making changes like this and committing the code; just be aware that if you regenerate, you'll overwrite those changes :)

Reply:
If you could point me to the code where those comparisons are taking place, I might be able to submit a fix.

Reply:
Hi @arajakul! I'm working on the compare stuff in this branch: https://github.com/jaypipes/aws-controllers-k8s/tree/delta

It is dependent on the refactoring currently going on in the code generator here:

https://github.com/aws/aws-controllers-k8s/pull/620

I expect to have the refactoring merged today and complete the compare resource changes early next week.